mtc.go

```diff
@@ -1489,6 +1588,84 @@ func (c *Claims) MarshalBinary() ([]byte, error) {
 	return b.Bytes()
 }
 
+func (e *Evidence) UnmarshalBinary(data []byte) error {
+	*e = Evidence{}
+	s := cryptobyte.String(data)
+	err := e.unmarshal(&s)
+	if err != nil {
+		return err
+	}
+	if !s.Empty() {
+		return ErrExtraBytes
+	}
+	return nil
+}
+
+func (e *Evidence) unmarshal(s *cryptobyte.String) error {
+
+	var (
+		evidenceInfo cryptobyte.String
+		evidenceType EvidenceType
+	)
+
+	if !s.ReadUint16((*uint16)(&evidenceType)) ||
+		!s.ReadUint24LengthPrefixed(&evidenceInfo) {
+		return ErrTruncated
+	}
+	e.Type = evidenceType
+	switch e.Type {
+	case EmptyEvidenceType:
+		if !evidenceInfo.Empty() {
+			return errors.New("non-empty info for empty evidence")
+		}
+	case X509ChainEvidenceType:
+		chain, err := x509.ParseCertificates([]byte(evidenceInfo))
```

Comment:
Given unmarshalling AssertionRequests is on the hot path of issuance, we might want to do lazy parsing here: just store the bytes in the struct and parse on demand. (As we do with the public key in TLSSubject)

Reply:
Makes sense. I'll address in a followup PR!